Classes/GarbageCollector.php

```diff
@@ -215,6 +219,33 @@ protected function deleteIndexDocuments($table, $uid)
     }
 
     /**
+     * Delete all index documents that are related to a TYPO3 "page"
+     *
+     * @param $uid
+     */
+    protected function deleteRecordsOfPage($uid)
```

Comment:
although the method name mentions records, you're also deleting Solr documents. That should be separated or named differently. 

Maybe you can also move this into `collectPageGarbage()`

There are methods here already that do what this method attempts to do; have a look at `deleteIndexDocuments()` and